### PR TITLE
Switch to new s1 preprocess graph, add support for .savedmodel file format

### PR DESCRIPTION
**Pull request by ScottMcMichael**

### CODE REVIEW

File: delta/extensions/metrics.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class SparseMetric(tensorflow.keras.metrics.Metric): # pylint:disable=abstract-m
     An abstract class for metrics applied to integer class labels,
     with networks that output one-hot encoding.
     """
-    def __init__(self, label, class_id: int=None, name: str=None, binary: int=False):
+    def __init__(self, label=None, class_id: int=None, name: str=None, binary: int=False, label_id: int=None, **kwargs):
```
Comment (bcoltin) [inline delta/extensions/metrics.py:33]: the class_id method should be able to take either a string or an integer, so probably this is unnecessary?

File: delta/extensions/sources/snap_process_sentinel1.sh [hunk 2]
```diff
@@ -22,9 +22,21 @@ export PATH=~/snap/bin:$PATH
 gptPath="gpt"
 
 # Get input parameters
-graphXmlPath="$1"
```
Comment (bcoltin) [inline delta/extensions/sources/snap_process_sentinel1.sh:25]: Can we move this and the sml file to scripts/convert? The delta folder should only be for python code

File: delta/ml/train.py [hunk 3]
```diff
@@ -389,23 +396,29 @@ def train(model_fn, dataset : ImageryDataset, training_spec, resume_path=None):
                                      callbacks=callbacks, verbose=1)
 
         if config.mlflow.enabled():
-            model_path = os.path.join(mcb.temp_dir, 'final_model.h5')
-            print('\nFinished, saving model to %s.' % (mlflow.get_artifact_uri() + '/final_model.h5'))
+            model_path = os.path.join(mcb.temp_dir, 'final_model' + config.train.default_model_extension())
```
Comment (bcoltin) [inline delta/ml/train.py:399]: I like how in the save function you checked if the extension was h5 and chose based on that. For the train command, an output filename can be given. Why don't instead of adding another field, we use whatever extension was given there? It could be an extra argument to the train function here, which in the train command we would extract from that filename if given. (just I expect the behavior of this option would be confusing)

File: delta/subcommands/classify.py [hunk 4]
```diff
@@ -128,11 +129,13 @@ def classify_image(model, image, label, path, net_name, options):
...
-            model = tf.keras.models.load_model(options.model, custom_objects=custom_objects(), compile=False)
-    else:
-        model = tf.keras.models.load_model(options.model, custom_objects=custom_objects(), compile=False)
+    #if config.general.gpus() == 0:
+    #    with tf.device('/cpu:0'):
+    #        model = tf.keras.models.load_model(options.model, custom_objects=custom_objects(), compile=False)
+    #else:
+    #    model = tf.keras.models.load_model(options.model, custom_objects=custom_objects(), compile=False)
+    model = load_model(options.model)
+    print('model.output_shape = ' + str(model.output_shape))
```
Comment (bcoltin) [inline delta/subcommands/classify.py:138]: remove print and comment

File: delta/extensions/sources/snap_process_sentinel1.sh (hunk 2 above)
Comment (bcoltin) [reply delta/extensions/sources/snap_process_sentinel1.sh:25]: Never mind, now I see why you did this! I would suggest now to leave the xml file, but get rid of the .sh and just have a system call in python (it is only one command). With my upcoming save_tiffs script having a standalone script should be less important. https://github.com/nasa/delta/pull/90/files#diff-43f7c73743790543ad7fbfeecbb6adc21d5ed70788a85e2203ea22086b381a9e

File: delta/extensions/sources/sentinel1.py [hunk 5]
```diff
@@ -115,8 +113,11 @@ def unpack_s1_to_folder(zip_path, unpack_folder):
             # - The SNAP tool *must* write to a .tif extension, so we have to
             #   rename the file if we want something else.
             temp_out_path = merged_path.replace('.vrt', '.tif')
-            cmd = (SNAP_SCRIPT_PATH + ' ' + SNAP_GRAPH_PATH + ' '
-                   + unpack_folder + ' ' + temp_out_path)
+            cmd = (SNAP_SCRIPT_PATH + ' ' + unpack_folder + ' ' + temp_out_path)
+            print(cmd)
+            os.system(cmd)
+            dimap_path = temp_out_path + '.dim'
+            cmd = 'pconvert -s 0,0 -f GeoTIFF-BigTiff -o ' + os.path.dirname(temp_out_path) +' '+ dimap_path
```
Comment (bcoltin) [inline delta/extensions/sources/sentinel1.py:120]: Is it documented somewhere how to install snap? Could you check if the programs are installed and given an error with a link if they aren't?

**Review by bcoltin — CHANGES_REQUESTED** [review]

**Comment by bcoltin** [comment]
Also can you fix the linter errors? Thanks!

File: delta/extensions/metrics.py (hunk 1 above)
Comment (ScottMcMichael) [reply delta/extensions/metrics.py:33]: Do you mean to just remove the int after class_id?

File: delta/ml/train.py (hunk 3 above)
Comment (ScottMcMichael) [reply delta/ml/train.py:399]: I saw this in subcommands/train.py:
```
if options.model is not None:
            save_model(model, options.model)
```
Is options.model always going to be available?  If so I can make this change.

File: delta/extensions/metrics.py (hunk 1 above)
Comment (bcoltin) [reply delta/extensions/metrics.py:33]: No, I meant remove the label_id argument entirely, because the label argument can already be a string or an int

File: delta/ml/train.py (hunk 3 above)
Comment (bcoltin) [reply delta/ml/train.py:399]: In subcommands/train.py it will always be available but may be None, so just choose the default for that case. You won't have access to options in delta/ml/train.py so I would add an argument to the train function there like save_format='h5'

File: delta/extensions/metrics.py (hunk 1 above)
Comment (ScottMcMichael) [reply delta/extensions/metrics.py:33]: Should I replace all references to self._label_id with self._class_id ?

Comment (ScottMcMichael) [reply delta/extensions/metrics.py:33]: If I make those changes then the model fails to load.  Maybe cleaning up label_id should be saved for another PR?

Comment (bcoltin) [reply delta/extensions/metrics.py:33]: Sorry if my last comment wasn't clear, I changed my mind and think it is ok as is. Can you just add label_id to the function documentation and mention it should only be used internally, and then I think we are good to go.

**Comment by bcoltin** [comment]
Sorry @ScottMcMichael, once last thing I realized we are missing--- can you modify scripts/model2config to work with the new format as well?

**Comment by ScottMcMichael** [comment]
> Sorry @ScottMcMichael, once last thing I realized we are missing--- can you modify scripts/model2config to work with the new format as well?

The config file copy is recorded in the .savedmodel folder as a plain text file so that part of the functionality is not needed.  Unfortunately the other half of it, the network printout, gives a weird error that could take a long time to diagnose so I don't know if we should hold up the PR over it.

**Comment by bcoltin** [comment]
Ok I'll make a new issue for it

**Review by bcoltin — APPROVED** [review]
